### PR TITLE
[IRGen] Lowered de/alloc_pack.

### PR DESCRIPTION
**Pull request by nate-chandler**
Lowered to dynamic or static allocation of `getPackElementSize() * count`.

### CODE REVIEW

**Comment by nate-chandler** [comment]
@swift-ci please test

**Comment by nate-chandler** [comment]
@swift-ci please test

File: lib/IRGen/GenPack.cpp [hunk 1]
```diff
@@ -587,15 +587,59 @@ void irgen::cleanupTypeMetadataPack(IRGenFunction &IGF,
...
+}
+
+StackAddress irgen::allocatePack(IRGenFunction &IGF, CanSILPackType packType) {
+  auto *shape = IGF.emitPackShapeExpression(packType);
+
+  auto elementSize = getPackElementSize(IGF.IGM, packType);
+
+  if (auto *constantInt = dyn_cast<llvm::ConstantInt>(shape)) {
+    assert(packType->getNumElements() == constantInt->getValue());
+    assert(!packType->containsPackExpansionType());
```
Comment (rjmccall) [inline lib/IRGen/GenPack.cpp:615]: You'll need a `(void) constantInt;` here to not get a warning in release builds.

File: test/IRGen/variadic_generics.sil [hunk 2]
```diff
@@ -74,35 +74,34 @@ bb0:
   return %index : $Builtin.PackIndex
 }
 
-// CHECK-LABEL: define{{.*}} @test_pack_element_get_1(
```
Comment (rjmccall) [inline test/IRGen/variadic_generics.sil:77]: I don't think you meant to delete my test case here. :)

**Review by rjmccall — COMMENTED** [review]
LGTM other than the test thing

File: test/IRGen/variadic_generics.sil (hunk 2 above)
Comment (nate-chandler) [reply test/IRGen/variadic_generics.sil]: Oops!

**Pull request merged**